components/scream/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -165,68 +168,33 @@ void SHOCMacrophysics::init_buffers(const ATMBufferManager &buffer_manager)
   m_buffer.pref_mid = decltype(m_buffer.pref_mid)(s_mem, nlev_packs);
   s_mem += m_buffer.pref_mid.size();
 
-  m_buffer.z_mid = decltype(m_buffer.z_mid)(s_mem, m_num_cols, nlev_packs);
-  s_mem += m_buffer.z_mid.size();
-  m_buffer.z_int = decltype(m_buffer.z_int)(s_mem, m_num_cols, nlevi_packs);
-  s_mem += m_buffer.z_int.size();
-  m_buffer.rrho = decltype(m_buffer.rrho)(s_mem, m_num_cols, nlev_packs);
-  s_mem += m_buffer.rrho.size();
-  m_buffer.rrho_i = decltype(m_buffer.rrho_i)(s_mem, m_num_cols, nlevi_packs);
-  s_mem += m_buffer.rrho_i.size();
-  m_buffer.thv = decltype(m_buffer.thv)(s_mem, m_num_cols, nlev_packs);
-  s_mem += m_buffer.thv.size();
-  m_buffer.dz = decltype(m_buffer.dz)(s_mem, m_num_cols, nlev_packs);
-  s_mem += m_buffer.dz .size();
-  m_buffer.zt_grid = decltype(m_buffer.zt_grid)(s_mem, m_num_cols, nlev_packs);
-  s_mem += m_buffer.zt_grid.size();
-  m_buffer.zi_grid = decltype(m_buffer.zi_grid)(s_mem, m_num_cols, nlevi_packs);
-  s_mem += m_buffer.zi_grid.size();
+  using spack_2d_view_t = decltype(m_buffer.z_mid);
+  spack_2d_view_t* _2d_spack_mid_view_ptrs[Buffer::num_2d_vector_mid] = {
```

Comment:
I took the opportunity to clean up some of the repetitive code in this method. The changes to the 1d code should be semantically identical to the previous impl. The 2d code does not behave exactly as before because the order of the views in the buffer is different with all the "mid" views and "int" views grouped separately now. I assume this order doesn't matter. If it does, I will refactor this to preserve the original order.

Reply:
@tcclevenger ^